boa/src/builtins/string/mod.rs

```diff
@@ -266,6 +267,39 @@ impl String {
         }
     }
 
+    /// `String.prototype.at ( index )`
+    ///
+    /// This String object's at() method returns a String consisting of the single UTF-16 code unit located at the specified position.
+    /// Returns undefined if the given index cannot be found.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/proposal-relative-indexing-method/#sec-string-prototype-additions
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/at
+    pub(crate) fn at(this: &Value, args: &[Value], context: &mut Context) -> Result<Value> {
+        let this = this.require_object_coercible(context)?;
+        let s = this.to_string(context)?;
+        let len = s.len();
+        let relative_index = args
+            .get(0)
+            .cloned()
+            .expect("No index provided")
+            .to_integer(context)?;
+        let k = if relative_index < 0 as f64 {
+            len - (-relative_index as usize)
+        } else {
+            relative_index as usize
+        };
+
+        if let Some(utf16_val) = s.encode_utf16().nth(k) {
+            Ok(Value::from(from_u32(utf16_val as u32).unwrap()))
```

Comment:
I would use `u32::from()` and change the `unwrap()` for an `expect()` in case something strange happens.
```suggestion
            Ok(Value::from(from_u32(u32::from(utf16_val)).expect("invalid utf-16 character")))
```

Reply:
Just wondering why is `u32::from()` preferred to safe casting?